journal.py

```diff
@@ -36,31 +43,70 @@ def list_view(request):
 
 @view_config(route_name='detail', renderer='templates/detail.jinja2')
 def detail_view(request):
-    article_id = request.matchdict['id']
-    article = Entry.get_article(article_id)
-    title = article[0].title
-    body_text = article[0].body_text
-    created = article[0].created
-    return {'title': title, 'body_text': body_text, 'created': created,
-            'article_id': article_id}
+    try:
+        article_id = request.matchdict['id']
+        article = Entry.get_article(article_id)
+        return {'article': article}
+    except NoResultFound:
+        return HTTPFound(request.route_url('home'))
 
 
 @view_config(route_name='new', renderer='templates/new.jinja2')
 def new_entry(request):
-    return {}
+    if request.method == 'GET':
+        return {}
+    if request.method == 'POST':
+        if request.authenticated_userid:
+            title = request.params.get('title')
+            body_text = request.params.get('body_text')
+            try:
+                newart = Entry.write(title=title, body_text=body_text)
+                DBSession.flush()
+            except ValueError:
+                return {'err_msg': 'Try Again: need both an entry and a title',
+                        'title': title,
+                        'body_text': body_text}
+            # TODO: edit points towards the detail page; new view probably
+            # should as well. Need to find a way to return the article id.
+            return HTTPFound(request.route_url('detail', id=newart.id))
+        else:
+            return HTTPForbidden()
+    else:
+        return HTTPMethodNotAllowed()
 
 
 @view_config(route_name='edit', renderer='templates/edit.jinja2')
-def new_entry(request):
-    return {}
-
-
-@view_config(route_name='add', request_method='POST')
-def add_entry(request):
-    title = request.params.get('title')
-    body_text = request.params.get('body_text')
-    Entry.write(title=title, body_text=body_text)
-    return HTTPFound(request.route_url('home'))
+def edit_entry(request):
+    if request.method == 'GET':
+        article_id = request.matchdict['id']
```

Comment:
You also do this again, below, in the block for POST requests.  That doesn't need to happen twice.  Why not get the ID regardless of the request method?  You need it in both places, no?